src/EnergyPlus/GroundTemperatureModeling/KusudaAchenbachGroundTemperatureModel.cc

```diff
@@ -0,0 +1,262 @@
+// C++ Headers
+#include<memory>
+
+// ObjexxFCL Headers
+
+// EnergyPlus Headers
+#include <DataIPShortCuts.hh>
+#include <GroundTemperatureModeling/KusudaAchenbachGroundTemperatureModel.hh>
+#include <GroundTemperatureModeling/GroundTemperatureModelManager.hh>
+#include <InputProcessor.hh>
+#include <WeatherManager.hh>
+
+namespace EnergyPlus {
+
+	//******************************************************************************
+
+	// Kusuda model factory
+	std::shared_ptr< KusudaGroundTempsModel > 
+	KusudaGroundTempsModel::KusudaGTMFactory( 
+		int objectType, 
+		std::string objectName
+	)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Matt Mitchell
+		//       DATE WRITTEN   Summer 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Reads input and creates instance of Kusuda ground temps model
+
+		// USE STATEMENTS:
+		using DataGlobals::SecsInDay;
+		using namespace DataIPShortCuts;
+		using namespace GroundTemperatureManager;
+
+		// Locals
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:	
+		bool found = false;
+		int NumNums;
+		int NumAlphas;
+		int IOStat;
+		bool ErrorsFound = false;
+
+		// New shared pointer for this model object
+		std::shared_ptr< KusudaGroundTempsModel > thisModel( new KusudaGroundTempsModel() );
+
+		std::string const cCurrentModuleObject = CurrentModuleObjects( objectType_KusudaGroundTemp );
+		int numCurrModels = InputProcessor::GetNumObjectsFound( cCurrentModuleObject );
+
+		for ( int modelNum = 1; modelNum <= numCurrModels; ++modelNum ) {
+
+			InputProcessor::GetObjectItem( cCurrentModuleObject, modelNum, cAlphaArgs, NumAlphas, rNumericArgs, NumNums, IOStat );
+
+			if ( objectName == cAlphaArgs( 1 ) ) {
+
+				// Read input into object here
+				thisModel->objectName = cAlphaArgs( 1 );
+				thisModel->objectType = objectType;
+				thisModel->groundThermalDiffisivity = rNumericArgs( 1 ) / ( rNumericArgs( 2 ) * rNumericArgs( 3 ) );
+
+				bool useGroundTempDataForKusuda = rNumericArgs( 4 ) || rNumericArgs( 5 ) || rNumericArgs( 6 );
+
+				if ( useGroundTempDataForKusuda ) {
+					// Use Kusuda Parameters
+					thisModel->aveGroundTemp = rNumericArgs( 4 );
+					thisModel->aveGroundTempAmplitude = rNumericArgs( 5 );
+					thisModel->phaseShiftInSecs = rNumericArgs( 6 ) * SecsInDay;
+				} else {
+					// Use data from Site:GroundTemperature:Shallow to generate parameters
+
+					int monthsInYear( 12 );
+					int avgDaysInMonth( 30 );
+					int monthOfMinSurfTemp( 0 );
+					Real64 averageGroundTemp( 0 );
+					Real64 amplitudeOfGroundTemp( 0 );
+					Real64 phaseShiftOfMinGroundTempDays( 0 );
+					Real64 minSurfTemp( 100 ); // Set high month 1 temp will be lower and actually get updated
+					Real64 maxSurfTemp( -100 ); // Set low initially but will get updated
+
+					std::shared_ptr< BaseGroundTempsModel > shallowObj = GetGroundTempModelAndInit( CurrentModuleObjects( objectType_SiteShallowGroundTemp ), "" );
```

Comment:
Ok I remembered (and did a little research). If you make the factory return unique_ptr instead of shared_ptr, you can 

```cpp
// converts std::unique_ptr to std::shared_ptr
std::shared_ptr< BaseGroundTempsModel > sp =
GetGroundTempModelAndInit( CurrentModuleObjects( objectType_SiteShallowGroundTemp ), "" );
```

But you can also then use the same factory method to generate unique_ptrs. 

That said, I now understand and see what you are doing. You are returning a shared_ptr from a vector< shared_ptr >, if already created. Hummm ok. The way you are doing it requires a shared_ptr so I think this is fine.

Slightly on topic segue, another option is to return std::weak_ptr instead of shared_ptr. This will not increase the reference count of the shared_ptr and is allowed to dangle. Read [here](http://en.cppreference.com/w/cpp/memory/weak_ptr) and [here](https://stackoverflow.com/questions/4984381/shared-ptr-and-weak-ptr-differences) for more information. I don't think we should do this for this PR, but is food for thought moving forward.